Classes from WAR project exported with `<attachClasses>true</attachClasses>` are not visible

### Expected behaviour
Classes (`my.MyClass`) from WAR project exported with option `<attachClasses>true</attachClasses>` and imported with
 ```
 <dependency>
     <groupId>testng-eclipse</groupId>
     <artifactId>my-war</artifactId>
     <version>0.0.1-SNAPSHOT</version>
     <type>jar</type>
     <classifier>classes</classifier>
</dependency>
```

should be visible when test is called by testng eclipse plugin

### Actual behaviour

> java.lang.NoClassDefFoundError: my/MyClass
	at my.MyTest.f(MyTest.java:8)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
(...)
Caused by: java.lang.ClassNotFoundException: my.MyClass
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)

### Steps to reproduce
1. Create WAR project with class my.MyClass and `<attachClasses>true</attachClasses>`
2. Create test project with dependency declared as above
3. Create test with instruction `new MyClass()`
4. Run test from Eclipse

### Running Configuration
* Plugin Version: 6.9.11.201604020423
* (Optional) Operating System: Win10

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -13,6 +13,7 @@ Supported Metrics:
 * enhancement #241: give more space to the testng result tree
 * issue #238: fixed the toolbar position on Eclipse Neon
 * fixed issue #157: NPE in TestRunnerViewPart
+* fixed #251: Classes from WAR project exported with `<attachClasses>true</attachClasses>` are not visible
 
 ## 6.9.11
 

--- a/testng-maven-eclipse-plugin/META-INF/MANIFEST.MF
+++ b/testng-maven-eclipse-plugin/META-INF/MANIFEST.MF
@@ -12,12 +12,14 @@ Require-Bundle: org.eclipse.ui,
  org.eclipse.m2e.core;bundle-version="1.5.0",
  org.eclipse.m2e.maven.runtime;bundle-version="1.5.0",
  org.eclipse.m2e.profiles.core;bundle-version="1.5.0",
+ org.eclipse.m2e.jdt;bundle-version="1.5.0",
  org.eclipse.core.variables,
  org.eclipse.ui.console,
  org.eclipse.debug.core,
  org.eclipse.debug.ui,
  org.testng.eclipse;bundle-version="6.9.10",
  org.eclipse.jdt.core,
+ org.eclipse.jdt.launching,
  org.eclipse.jdt.ui
 Bundle-ActivationPolicy: lazy
 Bundle-RequiredExecutionEnvironment: JavaSE-1.7

--- a/testng-maven-eclipse-plugin/plugin.xml
+++ b/testng-maven-eclipse-plugin/plugin.xml
@@ -33,5 +33,13 @@
              name="%org.testng.eclipse.maven.pref.name">
        </page>
     </extension>
+    <extension
+          point="org.eclipse.m2e.jdt.classifierClasspathProviders">
+       <classifierClasspathProvider
+             class="org.testng.eclipse.maven.WarClassesClassifierClasspathProvider"
+             id="org.testng.eclipse.maven.WarClassesClassifierClasspathProvider"
+             name="WarClassesClassifierClasspathProvider">
+       </classifierClasspathProvider>
+    </extension>
 
 </plugin>

--- a/testng-maven-eclipse-plugin/src/org/testng/eclipse/maven/WarClassesClassifierClasspathProvider.java
+++ b/testng-maven-eclipse-plugin/src/org/testng/eclipse/maven/WarClassesClassifierClasspathProvider.java
@@ -1,0 +1,41 @@
+package org.testng.eclipse.maven;
+
+import java.util.Set;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.launching.IRuntimeClasspathEntry;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
+import org.eclipse.m2e.jdt.AbstractClassifierClasspathProvider;
+
+public class WarClassesClassifierClasspathProvider extends AbstractClassifierClasspathProvider {
+
+  @Override
+  public boolean applies(IMavenProjectFacade mavenProjectFacade, String classifier) {
+    return getClassifier().equals(classifier);
+  }
+
+  @Override
+  public String getClassifier() {
+    return "classes";
+  }
+
+  /**
+   * Adds the main classes folder to the runtime classpath.
+   */
+  @Override
+  public void setRuntimeClasspath(Set<IRuntimeClasspathEntry> runtimeClasspath, IMavenProjectFacade mavenProjectFacade,
+      IProgressMonitor monitor) throws CoreException {
+    addMainFolder(runtimeClasspath, mavenProjectFacade, monitor);
+  }
+
+  /**
+   * Adds the main classes folder to the test classpath.
+   */
+  @Override
+  public void setTestClasspath(Set<IRuntimeClasspathEntry> testClasspath, IMavenProjectFacade mavenProjectFacade,
+      IProgressMonitor monitor) throws CoreException {
+    setRuntimeClasspath(testClasspath, mavenProjectFacade, monitor);
+  }
+
+}

